### ISSUE TITLE
How to modify requests module timeout when using corenlp?

### ISSUE THREAD

**Issue by ghost** [nltk/nltk#1917]
I want to parse long text using corenlp so the request tooks longer than the default timeout which is 60 seconds. Is it possible to change it?

![image](https://user-images.githubusercontent.com/12383137/34068405-d42d0a40-e274-11e7-9519-679967a1e086.png)

**Comment by dimazest** [comment]
Currently timeout is hardcoded in https://github.com/nltk/nltk/blob/develop/nltk/parse/corenlp.py#L254 so you need to change the source code.

**Comment by alvations** [comment]
@dimazest Maybe we'll need to expose the timeout argument when intializing the `CoreNLPParser()` 

But then again, we should be deprecating the module and directing people to the canonical Python wrapper https://pypi.python.org/pypi/stanford-corenlp/ ;P

**Comment by dimazest** [comment]
I agree, using the official bindings is the best solution in long term.

**Comment by programma256** [comment]
> 
> 
> @dimazest Maybe we'll need to expose the timeout argument when intializing the `CoreNLPParser()`
> 
> But then again, we should be deprecating the module and directing people to the canonical Python wrapper https://pypi.python.org/pypi/stanford-corenlp/ ;P

Just FYI the [version on github](https://github.com/stanfordnlp/python-stanford-corenlp) (3.92) has been updated in the last month to support returning json results via the API, so I think the functionality now exists to replace CoreNLPServer. v3.8 hasn't been undated on PyPI however so I suppose this is still in development.

For anyone wishing to parse long text via the CoreNLP API I suggest trying the official bindings first as it is more verbose about what is happening on the server and if the JVM runs out of memory. Just make sure you install from github. Using the NLTK interface the API call seemed to hang above a pretty modest number of input sentences and I couldn't fix it.

**Comment by khadkechetan** [comment]
```
### 
> Hi, 
> I have found the solution for that. 
> You can go to download path in your machine for NLTK.
import nltk
path =nltk.__file__
> go to the path directory and then nltk/parse folder. Open "corenlp.py" file and search for the following function.
## def api_call(self, data, properties=None, timeout=240):**
> Change the timeout value to your custom value. I have changed 60 to 240. I will work. I am able to solve the following error with this trick.
> -------------------------
ReadTimeout: HTTPConnectionPool(host='localhost', port=9000): Read timed out. (read timeout=60)
> -------------------------
> Thanks.
```
